x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.ts

```diff
@@ -0,0 +1,139 @@
+    index,
+    filterPath: ['aggregations.clusters.buckets'],
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          must: [
+            {
+              match: {
+                type: 'index_stats',
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: 'now-5m',
+                },
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        clusters: {
+          terms: {
+            include: clusters.map((cluster) => cluster.clusterUuid),
+            field: 'cluster_uuid',
+            size,
+          },
+          aggs: {
+            over_threshold: {
+              filter: {
+                range: {
+                  'index_stats.primaries.store.size_in_bytes': {
+                    gt: threshold * gbMultiplier,
+                  },
+                },
+              },
+              aggs: {
+                index: {
+                  terms: {
+                    field: 'index_stats.index',
+                    size,
+                  },
+                  aggs: {
+                    hits: {
+                      top_hits: {
+                        sort: [
+                          {
+                            timestamp: {
+                              order: 'desc',
+                              unmapped_type: 'long',
+                            },
+                          },
+                        ],
+                        _source: {
+                          includes: ['_index', 'index_stats', 'source_node'],
+                        },
+                        size: 1,
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const response = await callCluster('search', params);
+  const stats: IndexShardSizeStats[] = [];
+  const { buckets: clusterBuckets = [] } = response.aggregations.clusters;
+  const validIndexPatterns = memoizedIndexPatterns(shardIndexPatterns);
+
+  if (!clusterBuckets.length) {
+    return stats;
+  }
+
+  for (const clusterBucket of clusterBuckets) {
+    const indexBuckets = clusterBucket.over_threshold.index.buckets;
+    const clusterUuid = clusterBucket.key;
+
+    for (const indexBucket of indexBuckets) {
+      const shardIndex = indexBucket.key;
+      if (!ESGlobPatterns.isValid(shardIndex, validIndexPatterns)) {
+        continue;
+      }
+      const {
+        _index: monitoringIndexName,
+        _source: { source_node: sourceNode, index_stats: indexStats },
+      } = get(indexBucket, 'hits.hits.hits[0]');
+
+      const { size_in_bytes: shardSizeBytes } = indexStats.primaries.store;
+      const { name: nodeName, uuid: nodeId } = sourceNode;
+      const shardSize = +(shardSizeBytes / gbMultiplier).toFixed(2);
+      stats.push({
+        shardIndex,
+        shardSize,
+        clusterUuid,
+        nodeName,
```

Comment:
I'm seeing an error in the UI when I try and see the index listing page that I think is related to this:
<img width="797" alt="Screen Shot 2021-01-28 at 12 54 40 PM" src="https://user-images.githubusercontent.com/56682/106292337-b4632400-621a-11eb-9ddb-30c86e16bf75.png">